src/transformStateWithClones.js

```diff
@@ -7,7 +7,59 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesHistory = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    stateCopy = { ...stateCopy };
+
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = addProperties(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        stateCopy = removeProperties(stateCopy, action.keysToRemove);
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+
+      default:
+        throw new Error('Unknown action type!');
```

Comment:
[CODE STYLE]: Use a `default` case in your `switch` statement for error handling to ensure that any unexpected `action.type` values are caught. This way, you can throw an error or handle it accordingly.